File: backend/laravel-12.x/flutter_app/lib/models/crypto_pair.dart
class CryptoPair {
  final int id;
  final String symbol;
  final String baseCurrency;
  final String quoteCurrency;
  final double currentPrice;
  final double priceChange24h;
  final double priceChangePercentage24h;
  final double high24h;
  final double low24h;
  final double volume24h;
  final double? marketCap;
  final bool isActive;
  final DateTime lastUpdated;

  CryptoPair({
    required this.id,
    required this.symbol,
    required this.baseCurrency,
    required this.quoteCurrency,
    required this.currentPrice,
    required this.priceChange24h,
    required this.priceChangePercentage24h,
    required this.high24h,
    required this.low24h,
    required this.volume24h,
    this.marketCap,
    required this.isActive,
    required this.lastUpdated,
  });

  factory CryptoPair.fromJson(Map<String, dynamic> json) {
    return CryptoPair(
      id: json['id'],
      symbol: json['symbol'],
      baseCurrency: json['base_currency'],
      quoteCurrency: json['quote_currency'],
      currentPrice: double.parse(json['current_price'].toString()),
      priceChange24h: double.parse(json['price_change_24h'].toString()),
      priceChangePercentage24h: double.parse(json['price_change_percentage_24h'].toString()),
      high24h: double.parse(json['high_24h'].toString()),
      low24h: double.parse(json['low_24h'].toString()),
      volume24h: double.parse(json['volume_24h'].toString()),
      marketCap: json['market_cap'] != null 
          ? double.parse(json['market_cap'].toString()) 
          : null,
      isActive: json['is_active'] ?? true,
      lastUpdated: DateTime.parse(json['last_updated']),
    );
  }

  String get formattedPrice => '\$${currentPrice.toStringAsFixed(8)}';
  String get formattedVolume => '\$${(volume24h / 1000000).toStringAsFixed(1)}M';
  String get formattedChange => '${priceChangePercentage24h >= 0 ? '+' : ''}${priceChangePercentage24h.toStringAsFixed(2)}%';
  
  bool get isPositiveChange => priceChangePercentage24h >= 0;
}

File: backend/laravel-12.x/app/Models/CryptoPair.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CryptoPair extends Model
{
    use HasFactory;

    protected $fillable = [
        'symbol',
        'base_currency',
        'quote_currency',
        'current_price',
        'price_change_24h',
        'price_change_percentage_24h',
        'high_24h',
        'low_24h',
        'volume_24h',
        'market_cap',
        'is_active',
        'last_updated',
    ];

    protected $casts = [
        'current_price' => 'decimal:8',
        'price_change_24h' => 'decimal:4',
        'price_change_percentage_24h' => 'decimal:4',
        'high_24h' => 'decimal:8',
        'low_24h' => 'decimal:8',
        'volume_24h' => 'decimal:8',
        'market_cap' => 'decimal:2',
        'is_active' => 'boolean',
        'last_updated' => 'datetime',
    ];

    public function trades()
    {
        return $this->hasMany(Trade::class, 'pair', 'symbol');
    }

    public function orders()
    {
        return $this->hasMany(Order::class, 'pair', 'symbol');
    }

    public function getFormattedPriceAttribute()
    {
        return number_format($this->current_price, 8);
    }

    public function getFormattedVolumeAttribute()
    {
        return number_format($this->volume_24h, 2);
    }

    public function getPriceChangeColorAttribute()
    {
        return $this->price_change_percentage_24h >= 0 ? 'green' : 'red';
    }

    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    public function scopePopular($query)
    {
        return $query->orderBy('volume_24h', 'desc');
    }
}
